core/crypto/crypto.c

```diff
@@ -18,9 +18,10 @@ TEE_Result crypto_hash_alloc_ctx(void **ctx __unused, uint32_t algo __unused)
 	return TEE_ERROR_NOT_IMPLEMENTED;
 }
 
-void crypto_hash_free_ctx(void *ctx __unused, uint32_t algo __unused)
+void crypto_hash_free_ctx(void *ctx, uint32_t algo __unused)
```

Comment:
`crypto_hash_free_ctx()` in `core/lib/libtomcrypt/src/tee_ltc_provider.c` also need this fix. Same for `crypto_cipher_free_ctx()` and `crypto_mac_free_ctx()`

Reply:
How would they fail if `ctx == NULL`?

Reply:
The assert here could fail
https://github.com/OP-TEE/optee_os/blob/master/core/lib/libtomcrypt/src/tee_ltc_provider.c#L408
if `crypto_hash_alloc_ctx()` returns `TEE_ERROR_NOT_SUPPORTED`